source/Calamari.Tests/KubernetesFixtures/KubernetesContextScriptWrapperLiveFixtureBase.cs

```diff
@@ -104,6 +165,52 @@ protected void TestScript(IScriptWrapper wrapper, string scriptName)
             }
         }
 
+        protected void DeployWithScriptAndVerifySuccess(IReadOnlyList<IScriptWrapper> wrappers,
+            ICalamariFileSystem fileSystem, Action<TemporaryDirectory> addFilesAction = null)
+        {
+            SetupTempDirectoryAndVerifyResult(addFilesAction, () =>
+            {
+                var scriptPath = Path.Combine(testFolder, "KubernetesFixtures/Scripts");
+                variables.Set(SpecialVariables.Action.Script.ScriptBodyBySyntax(ScriptSyntax.Bash),
+                    File.ReadAllText(Path.Combine(scriptPath, "KubernetesDeployment.sh")));
+                variables.Set(SpecialVariables.Action.Script.ScriptBodyBySyntax(ScriptSyntax.PowerShell),
+                    File.ReadAllText(Path.Combine(scriptPath, "KubernetesDeployment.ps1")));
+
+                return ExecuteScript(wrappers, null, fileSystem);
+            });
+        }
+
+        private void SetupTempDirectoryAndVerifyResult(Action<TemporaryDirectory> addFilesAction, Func<CalamariResult> func)
+        {
+            using (var dir = TemporaryDirectory.Create())
+            {
+                var folderPath = Path.Combine(dir.DirectoryPath, "TestFolder");
+                Directory.CreateDirectory(folderPath);
+                var currentDirectory = Environment.CurrentDirectory;
+                Environment.CurrentDirectory = folderPath;
```

Comment:
Is there any other way we can write to the temporary path? Or do the scripts use Environment.CurrentDirectory?

Reply:
I have figured out why it wasn't working. There are placed in the code where we use `Path.GetFilePath()` on a relative path so it uses the `Environment.CurrentDirectory` which is obviously not right if we're working in a different directory. I've updated the areas that affect this test.